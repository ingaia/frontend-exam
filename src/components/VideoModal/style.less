@import './../../variables.less';

.video-modal {
    display: flex;
    justify-content: center;
    align-items: center;
    position: fixed; 
    left: 0;
    top: 0;
    width: 100%; 
    height: 100%; 
    background: rgba(29, 26, 26, 0.9);
    z-index: 20; 

    .video-content {
        padding: 10px;
        background-color: #000;
        border: 2px solid @hillary;

        iframe {
            border: none;
        }
    }

    .btn-default {
        position: absolute;
        right: 10px;
        top: 10px;
    }
}

@media @mobile {
    .video-modal {
        .video-content {
            width: 100%;
            iframe {
                width: 100%;
                height: 350px;
            }
        }
    }
}