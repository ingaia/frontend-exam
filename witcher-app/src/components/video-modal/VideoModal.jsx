import PropTypes from 'prop-types';
import React from 'react';

import styles from './VideoModal.module.scss';

const Modal = (props) => {

    const { videoId, onClose } = props;

    return (
        !videoId
            ? null
            : <div className={styles.ModalOverlay}>

                <div className={styles.ModalClose} onClick={onClose}>CLOSE</div>

                <div className={styles.Modal}>
                    <iframe 
                        title="video"
                        src={`https://www.youtube.com/embed/${videoId}?start=1`} 
                        frameBorder="0" 
                        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" 
                        allowFullScreen>
                    </iframe>
                </div>
            </div>
    );
}
 
Modal.propTypes = {
    isShow: PropTypes.bool,
    videoId: PropTypes.string,
    onClose: PropTypes.func.isRequired,
}

Modal.defaultProps = {
    isShow: false,
}

export default Modal;