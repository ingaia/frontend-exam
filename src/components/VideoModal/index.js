import React from 'react';

import './style.less';
import Button from '../Button';

export default function VideoModal(props) {
    return (
        <div className="video-modal">
            <Button className="btn-default" label="Close" onClick={props.closeVideo}/>
            <div className="video-content">
                <iframe width="900" height="530" src={`https://www.youtube.com/embed/${props.videoId}?autoplay=1&showinfo=0` }>
                </iframe>
            </div>
        </div>
    );
}