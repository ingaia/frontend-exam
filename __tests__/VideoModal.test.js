import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import VideoModal from '../src/components/VideoModal';

describe('<VideoModal/>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<VideoModal/>);
    });

    it('should render', () => {
        expect(wrapper.hasClass('video-modal')).toBe(true);
        expect(wrapper.find('iframe').length).toBe(1);
        expect(wrapper.find('Button').length).toBe(1);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});