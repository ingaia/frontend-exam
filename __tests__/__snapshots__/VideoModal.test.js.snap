// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoModal/> should render 1`] = `
<div
  className="video-modal"
>
  <Button
    className="btn-default"
    label="Close"
  />
  <div
    className="video-content"
  >
    <iframe
      height="530"
      src="https://www.youtube.com/embed/undefined?autoplay=1&showinfo=0"
      width="900"
    />
  </div>
</div>
`;
